messaging/broker.go

```diff
@@ -165,6 +185,22 @@ func (b *Broker) Open(path string) error {
 			return fmt.Errorf("open topics: %s", err)
 		}
 
+		// Start topic truncation.
+		b.wg.Add(1)
+		go func() {
+			defer b.wg.Done()
+			tick := time.NewTicker(b.TruncationInterval)
+			defer tick.Stop()
+			for {
+				select {
+				case <-tick.C:
+					b.TruncateTopics()
+				case <-b.done:
+					return
+				}
+			}
+		}()
```

Comment:
Can you move this to a named function so it's isolated? Also, the `Open()` is already kind of big.